src/Authentication/Authenticators/Session.php

```diff
@@ -655,15 +655,17 @@ public function logout(): bool
 
         // Destroy the session data - but ensure a session is still
         // available for flash messages, etc.
-        if (isset($_SESSION)) {
-            foreach (array_keys($_SESSION) as $key) {
-                $_SESSION[$key] = null;
-                unset($_SESSION[$key]);
+        /** @var \CodeIgniter\Session\Session $session */
+        $session     = session();
+        $sessionData = $session->get();
+        if (isset($sessionData)) {
```

Comment:
I believe `get()` without a parameter always returns an array - this should be unnecessary.